install.sh

```diff
@@ -71,22 +115,64 @@ main () {
         exit 1
     fi
 
-    URL="https://github.com/opengrep/opengrep/releases/download/${VERSION}/${DIST}"
-    echo
-    echo "*** Installing Opengrep ${VERSION} for ${OS} (${ARCH}) ***"
-    echo
+    if [ "$REQUESTED_VERSION" = "latest" ]; then
+        # Use GitHub's latest release redirect
+        URL="https://github.com/opengrep/opengrep/releases/latest/download/${DIST}"
+
+        # Get the final version from the redirect
+        REDIRECT_URL=$(curl -sI "$URL" 2>/dev/null | grep -i '^location:' | sed 's/location: //i' | tr -d '\r\n' || true)
+        if [ -n "$REDIRECT_URL" ]; then
+            # Extract version from the redirect URL
+            FINAL_VERSION=$(echo "$REDIRECT_URL" | sed -E 's|.*/download/([^/]+)/.*|\1|')
+        else
+            echo "Failed to resolve latest version. The redirect from $URL did not work." 1>&2
+            echo "Please check your internet connection or specify a version with -v" 1>&2
+            exit 1
+        fi
+
+        echo
+        echo "*** Installing Opengrep ${FINAL_VERSION} for ${OS} (${ARCH}) ***"
+        echo
+    else
+        URL="https://github.com/opengrep/opengrep/releases/download/${REQUESTED_VERSION}/${DIST}"
+        echo
+        echo "*** Installing Opengrep ${REQUESTED_VERSION} for ${OS} (${ARCH}) ***"
+        echo
+    fi
+
+    # Now set INST with the final version
+    INST="${PREFIX}/${FINAL_VERSION}"
 
     # check if binary already exists
     if [ -f "${INST}/opengrep" ]; then
         echo "Destination binary ${INST}/opengrep already exists."
     else
+        # Download to temp file first
+        TEMP_FILE=$(mktemp)
+        trap 'rm -f $TEMP_FILE' EXIT
+
+        if ! retry curl --fail --location --progress-bar "${URL}" > "${TEMP_FILE}"; then
```

Comment:
Moving the conversation here:
[Comment 1](https://github.com/opengrep/opengrep/pull/307#issuecomment-3002033389)

> As a first note, if the requested version is not found, then why retry?
> 
> Will review in more detail as soon as possible.

[Comment 2](https://github.com/opengrep/opengrep/pull/307#issuecomment-3005095383)

> @dimitris-m because sometimes the failure is due to a GitHub rate limit / failure, not because the release isn't actually there.

[Comment 3](https://github.com/opengrep/opengrep/pull/307#issuecomment-3005259747)

> But in case of rate limit, what is the point of immediately retrying?
>
> I'd rather have the http code inform the error message and fail in such cases, which I assume to be infrequent.

Reply:
I've been using this script for about a day and I've hit this problem when building in a GitHub Action pipelines a small handful of times (I think maybe 4 or 5 times). This added retry has succeeded ever since. It's a small sample size, but I'm willing to bet something as simple as `curl --fail || curl --fail` would even help a great bit, but having the 2 second delay seemed prudent

Reply:
but it's enough to repeat until success, if you know that the version exists.
especially without `-v <version>`, it should only fail on connectivity issues (modulo `--verify-signatures`).